File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/python/shapes.py
from dataclasses import dataclass
from typing import Protocol


# BEGIN PROBE F14.P3
class Shape(Protocol):
    def area(self) -> float: ...


@dataclass
class Circle:
    r: float

    def area(self) -> float:
        return 3.141592653589793 * self.r * self.r


@dataclass
class Rect:
    w: float
    h: float

    def area(self) -> float:
        return self.w * self.h
# END PROBE F14.P3

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/shapes/shapes.go
package shapes

// BEGIN PROBE F14.P3
type Shape interface{ Area() float64 }

type Circle struct{ r float64 }

func (c Circle) Area() float64 { return 3.141592653589793 * c.r * c.r }

type Rect struct{ w, h float64 }

func (r Rect) Area() float64 { return r.w * r.h }

// END PROBE F14.P3
